modules/standard/Set.chpl

```diff
@@ -284,9 +284,15 @@ module Set {
         var (isFullSlot, idx) = _htb.findAvailableSlot(elem);
 
         if !isFullSlot {
+
+          // This line moves the bits over, 'elem' is dead past this point.
```

Comment:
I am a little confused by this comment. By "dead" do you mean that it is no longer needed, or does `moveToValue` deallocate `elem`'s memory? It must be the latter, otherwise we'd leak a lot, right? The word "dead" makes me think that it is the former, though.

Reply:
Ah, what I was trying to express is that `moveToValue` consumes the bits of `elem` such that after the call `elem` refers to stale memory. There's no deallocation though.

How about `This line moves the bits over, 'elem' cannot be used past this point`?

Reply:
I am a bit confused, then. But this code doesn't leak as far as we know, so I think I am missing something.

`elem` has `in` intent and `no auto destroy`. It gets freed upon successful addition... how? Hmm... moveToValue has a const ref argument, so maybe this becomes a pointer assignment. I think that'd be convincing..

In any case, I am fine with this comment, then.